nbextensions/bulkImportCell/tabs/xsvGenerator.js

```diff
@@ -45,25 +139,39 @@ define(['StagingServiceClient', 'common/html', 'common/runtime', 'common/ui', 'u
 
         async run() {
             const body = this.renderLayout();
-            const confirmed = await UI.showConfirmDialog({
+            const modalArgs = {
                 title: 'Create CSV Template',
                 body,
-                okLabel: 'Create Template',
+                okLabel: 'Generate template!',
+                doThisFirst: (mdn) => {
+                    const submit = mdn.querySelector('[data-element="ok"]');
+                    // TODO: the form should monitor its state of validity and
+                    // only allow the query to be submitted when the form is valid
+                    submit.addEventListener('click', this.runRequest.bind(this));
```

Comment:
temp code -- the finished implementation should use a params widget like the configure tab or appParamsWidget does